src/python/foglamp/data_purge/purge.py

```diff
@@ -0,0 +1,332 @@
+
+
+def execute_command_with_return_value(stmt: str) -> dict:
+    """Imitate connection to postgres that returns result.    
+    Args:
+        stmt (str): generated SQL query   
+    Returns:
+        Returns result set 
+    """
+    query_result = _CONN.execute(stmt)
+    return query_result.fetchall()
+
+
+def execute_command_without_return_value(stmt: str) -> None:
+    """Imitate connection to Postgres and a query that doesn't generate results
+    Args:
+        stmt (str): DELETE stmt 
+    """
+    _CONN.execute(stmt)
+    _CONN.execute("commit")
+
+
+def set_id() -> int:
+    """
+    Set the ID value for the next purge log table
+    Args: 
+        table that will store id
+    Returns:
+        Next INT in purge logging table
+    """
+    stmt = sqlalchemy.select([_PURGE_LOGGING_TABLE.c.id]).select_from(_PURGE_LOGGING_TABLE).order_by(
+        _PURGE_LOGGING_TABLE.c.id.desc()).limit(1)
+    result = execute_command_with_return_value(stmt)
+    if not result:
+        return 1
+    return int(result[0][0])+1
+
+
+"""The actual purge process 
+"""
+
+
+def purge(config) -> None:
+    """The actual process read the configuration file, and based off the information in it does the following:
+    1. Gets previous information found in log file
+    2. Based on the configurations, call the DELETE command to purge the data
+    3. Calculate relevant information kept in logs
+    4. Based on the configuration calculates how long to wait until next purge, and returns that 
+         
+    Returns:
+        Amount of time until next purge process
+    """
+
+    table_name = _READING_TABLE  # This could be replaced with any table that would need to be purged.
+
+    start_time = datetime.datetime.fromtimestamp(time.time())
+
+    age_timestamp = datetime.datetime.strftime(start_time - convert_timestamp(
+        set_time=config['age']['value']), '%Y-%m-%d %H:%M:%S.%f')
+    start_time = start_time.strftime('%Y-%m-%d %H:%M:%S')
+
+    # Number of unsent rows
+    number_sent_rows_query = sqlalchemy.select([sqlalchemy.func.count()]).select_from(table_name).where(
+        table_name.c.ts <= start_time).where(table_name.c.id > _LAST_ID)
+
+    unsent_rows_before = execute_command_with_return_value(number_sent_rows_query)
+    unsent_rows_before = int(unsent_rows_before[0][0])
+
+    """Time purge process starts
+    If unsent data is retained, then the WHERE condition is against the last sent ID
+    """
+
+    if config['retainUnsent']['value'] == 'True':
+        count_query = sqlalchemy.select([sqlalchemy.func.count()]).select_from(table_name).where(
+            table_name.c.id <= _LAST_ID).where(table_name.c.ts <= start_time)
+        total_count_before = execute_command_with_return_value(count_query)
+        total_count_before = int(total_count_before[0][0])
+
+        delete_query = sqlalchemy.delete(table_name).where(table_name.c.id <= _LAST_ID).where(
+            table_name.c.ts <= start_time)
+
```

Comment:
To get the number of rows deleted:
    result = _CONN.execute(delete_query)
    query_result.rowcount 
will give you the rows deleted, you need not query DB again to see the rows deleted